vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewState.kt

```diff
@@ -19,11 +19,26 @@ package im.vector.app.features.settings.devices.v2
 import com.airbnb.mvrx.Async
 import com.airbnb.mvrx.MavericksState
 import com.airbnb.mvrx.Uninitialized
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterType
+import org.matrix.android.sdk.api.extensions.orFalse
 
 data class DevicesViewState(
         val currentSessionCrossSigningInfo: CurrentSessionCrossSigningInfo = CurrentSessionCrossSigningInfo(),
         val devices: Async<List<DeviceFullInfo>> = Uninitialized,
         val unverifiedSessionsCount: Int = 0,
         val inactiveSessionsCount: Int = 0,
         val isLoading: Boolean = false,
-) : MavericksState
+        val currentFilter: DeviceManagerFilterType = DeviceManagerFilterType.ALL_SESSIONS,
+) : MavericksState {
+
+    fun List<DeviceFullInfo>?.filteredDevices(): List<DeviceFullInfo>? {
```

Comment:
What do you think about removing this extension method and move this logic inside a dedicated use case? The new use case could be called inside the `observeDevices()` method in the `ViewModel`. This way we can add unit tests for the new use case and for the change in the `ViewModel`.

Reply:
Was hard to refactor but done :)